test/Microsoft.AspNetCore.Razor.Test/Parser/CSharp/CSharpToMarkupSwitchTest.cs

```diff
@@ -553,6 +553,24 @@ public void ParseBlockParsesMarkupStatementOnSwitchCharacterFollowedByColon()
         }
 
         [Fact]
+        public void ParseBlockParsesMarkupStatementOnSwitchCharacterFollowedByDoubleColon()
```

Comment:
Add a few more tests.

1. Triple colon. The tokenizer special cases double colon so we want to make sure triple doesn't cause any issues.
1. A code generation test that has the double and triple colon in the markup. You can add those two lines [here](https://github.com/aspnet/Razor/blob/173886bf6cda465e839b60d262faf7f614c3b2ec/test/Microsoft.AspNetCore.Razor.Test/TestFiles/CodeGenerator/Source/Templates.cshtml#L22). To regenerate the corresponding C# files delete the `__RemoveThisBitTo__` text from [this](https://github.com/aspnet/Razor/blob/dev/test/Microsoft.AspNetCore.Razor.Test/project.json#L37) file. Run `./build verify` at the root of the project.